tests/request_parsing_tests.rs

```diff
@@ -0,0 +1,187 @@
+use std::borrow::Cow;
+
+use x11rb::errors::ParseError;
+use x11rb::x11_utils::RequestHeader;
+
+#[test]
+fn test_bad_request_header_opcode() {
```

Comment:
If I understand this correctly: Parsing should fail because the opcode is wrong. However, this could also fail to parse because of missing data. Would it be better to give this a correct request, e.g. `NoOperation` that can then only fail because of the opcode?

Reply:
So you'd prefer this be an entirely valid request except for the opcode? ok

Reply:
Yes. Since `ParseError` does not differentiate between "not enough bytes" and "some value was wrong", the test could otherwise succeed even though it should fail.

Reply:
Done.